core/src/main/scala/com/mesosphere/usi/core/javaapi/SchedulerAdapter.scala

```diff
@@ -0,0 +1,87 @@
+package com.mesosphere.usi.core.javaapi
+
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import akka.{Done, NotUsed}
+import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult, scaladsl}
+import akka.stream.javadsl.{Flow, Sink, SinkQueueWithCancel, Source, SourceQueueWithComplete}
+import com.mesosphere.usi.core.Scheduler.{SpecInput, StateOutput}
+import com.mesosphere.usi.core.scalaapi.{SchedulerAdapter => ScalaSchedulerAdapter}
+import com.mesosphere.usi.core.models.{SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+
+import scala.compat.java8.FutureConverters._
+import akka.japi.tuple.Tuple3
+import akka.japi.Pair
+
+import scala.concurrent.ExecutionContext
+
+class SchedulerAdapter(
+    schedulerFlow: scaladsl.Flow[SpecInput, StateOutput, NotUsed],
+    mat: Materializer,
+    ec: ExecutionContext) {
+
+  private implicit val materializer: Materializer = mat
+  private implicit val executionContext: ExecutionContext = ec
+
+  private val delegate = new ScalaSchedulerAdapter(schedulerFlow)
+
+  /**
+    * Represents the scheduler as a Sink and Source.
+    *
+    * This method will materialize the scheduler first, then Sink and Source can be materialized independently.
+    *
+    * @param specsSnapshot Snapshot of the current specs
+    * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
+    */
+  def asSourceAndSink(specsSnapshot: SpecsSnapshot = SpecsSnapshot.empty)
```

Comment:
Defaults don't work for Java, you'll have to use method overloading. At least I think so.

Reply:
I don't think it's necessary to write it in Java, Scala classes are totally usable from Java if we follow a few principles and don't overuse scala features.

Reply:
I'd really like to see some Java unit tests that use this